examples/hello_world/module/rust/src/lib.rs

```diff
@@ -18,31 +18,65 @@ pub mod proto {
     include!(concat!(env!("OUT_DIR"), "/oak.examples.hello_world.rs"));
 }
 
+use either::Either;
 use log::info;
-use oak::{grpc, io::ReceiverExt};
+use oak::grpc;
 use oak_abi::{label::Label, proto::oak::application::ConfigMap};
+use oak_io::Sender;
 use proto::{HelloRequest, HelloResponse, HelloWorld, HelloWorldDispatcher};
 
-oak::entrypoint!(oak_main<ConfigMap> => |_receiver| {
-    oak::logger::init_default();
-    let node = Node {
-        translator: grpc::client::Client::new(
+#[derive(Default)]
+struct Main;
+
+oak::entrypoint_command_handler!(oak_main => Main);
+
+impl oak::CommandHandler for Main {
+    type Command = ConfigMap;
+
+    fn handle_command(&mut self, _command: Self::Command) -> anyhow::Result<()> {
+        let translator_sender_result = oak::io::node_create::<grpc::Invocation>(
             "translator",
+            &Label::public_untrusted(),
             &oak::node_config::wasm("translator", "oak_main"),
-            &Label::public_untrusted()
-        )
-        .map(translator_common::TranslatorClient),
-    };
-    let dispatcher = HelloWorldDispatcher::new(node);
-    let grpc_channel =
-        oak::grpc::server::init("[::]:8080").expect("could not create gRPC server pseudo-Node");
-    oak::run_command_loop(dispatcher, grpc_channel.iter());
-});
+        );
+        let handler_init_sender = oak::io::entrypoint_node_create::<HelloWorldDispatcher<Node>>(
+            "handler",
+            &Label::public_untrusted(),
+            "app",
+        )?;
+        let handler_command_sender = oak::io::send_init(
+            handler_init_sender,
+            translator_sender_result
```

Comment:
nit: I know it is basically the same, but this feels clunky and does not read easily. It might be worth implementing `Encodable` and `Decodable` for `Option<T>`, especially if this (having a sender that might or might not exist) is a pattern that might be happen regularly.

Reply:
I agree, I will look into it next.

Reply:
(though I don't think it will happen often TBH)